tests/unit/recommenders/evaluation/test_python_time.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+import pandas as pd
+import pytest
+from unittest.mock import Mock
+from sklearn.preprocessing import minmax_scale
+
+from recommenders.utils.constants import (
+    DEFAULT_USER_COL,
+    DEFAULT_ITEM_COL,
+    DEFAULT_RATING_COL,
+    DEFAULT_PREDICTION_COL,
+    SEED,
+)
+from recommenders.evaluation.python_evaluation import (
+    merge_rating_true_pred,
+    merge_ranking_true_pred,
+    rmse,
+    mae,
+    rsquared,
+    exp_var,
+    get_top_k_items,
+    precision_at_k,
+    recall_at_k,
+    ndcg_at_k,
+    map_at_k,
+    auc,
+    logloss,
+)
+import random
+from recommenders.utils.timer import Timer
+
+random.seed(SEED)
+DATA_USER_NUM = 10000
+DATA_ITEM_NUM = DATA_USER_NUM * 2
+DATA_SAMPLE_NUM = DATA_USER_NUM * 1000
+DATA_RATING_MAX = 5
+
+# fmt: off
+DF_RATING_TRUE = pd.DataFrame(
+    {
+        DEFAULT_USER_COL: random.choices(range(0, DATA_USER_NUM), k=DATA_SAMPLE_NUM),
+        DEFAULT_ITEM_COL: random.choices(range(0, DATA_ITEM_NUM), k=DATA_SAMPLE_NUM),
+        DEFAULT_RATING_COL: random.choices(range(1, DATA_RATING_MAX+1), k=DATA_SAMPLE_NUM),
+    }
+)
+
+
+DF_RATING_PRED = pd.DataFrame(
+    {
+        DEFAULT_USER_COL: random.choices(range(0, DATA_USER_NUM), k=DATA_SAMPLE_NUM),
+        DEFAULT_ITEM_COL: random.choices(range(0, DATA_ITEM_NUM), k=DATA_SAMPLE_NUM),
+        DEFAULT_PREDICTION_COL: random.choices(range(1, DATA_RATING_MAX+1), k=DATA_SAMPLE_NUM),
+        # DEFAULT_RATING_COL: random.choices(range(1, RATING_MAX+1), k=SAMPLE_NUM),
+    }
+)
+
+
+DF_RATING_NOHIT = pd.DataFrame(
+    {
+        DEFAULT_USER_COL: random.choices(range(0, DATA_USER_NUM), k=DATA_SAMPLE_NUM),
+        DEFAULT_ITEM_COL: [DATA_ITEM_NUM] * DATA_SAMPLE_NUM,
+        DEFAULT_PREDICTION_COL: random.choices(range(1, DATA_RATING_MAX+1), k=DATA_SAMPLE_NUM),
+    }
+)
+
+
+@pytest.fixture
+def rating_true():
+    return DF_RATING_TRUE.copy()
+
+
+@pytest.fixture
+def rating_pred():
+    return DF_RATING_PRED.copy()
+
+
+@pytest.fixture
+def rating_nohit():
+    return DF_RATING_NOHIT.copy()
+# fmt: on
+
+
+@pytest.fixture
+def rating_true_binary(rating_true):
+    # Convert true ratings to binary
+    rating_true[DEFAULT_RATING_COL] = rating_true[DEFAULT_RATING_COL].apply(
+        lambda x: 1.0 if x >= 3 else 0.0
+    )
+    return rating_true
+
+
+@pytest.fixture
+def rating_pred_binary(rating_pred):
+    # Normalize the predictions
+    rating_pred[DEFAULT_PREDICTION_COL] = minmax_scale(
+        rating_pred[DEFAULT_PREDICTION_COL].astype(float)
+    )
+    return rating_pred
+
+
+# The following time thresholds are benchmarked on Azure
+# Standard_D14_v2 VM, with Intel(R) Xeon(R) CPU E5-2673 v3 @ 2.40GHz,
+# 16 cores, 112 GB RAM, 800 GB disk.
+# The thresholds are calculated by MEAN + 5 * STANDARD DEVIATION.
+
+
+def test_merge_rating(rating_true, rating_pred):
```

Comment:
@pradnyeshjoshi Chuyang needs to add these tests to some of the test groups. I'm thinking that we might want to create a small tutorial to help developers add tests. What do you think?